pkg/datapath/iptables/iptables.go

```
@@ -267,46 +275,73 @@ type Manager struct {
 	CNIChainingMode      string
 }
 
-// Init initializes the iptables manager and checks for iptables kernel modules
-// availability.
-func (m *Manager) Init(modulesManager *modules.Manager) {
-	haveIp6tables := true
-	if err := modulesManager.FindOrLoadModules(
-		"ip_tables", "iptable_nat", "iptable_mangle", "iptable_raw",
-		"iptable_filter"); err != nil {
-		log.WithError(err).Warning(
+type params struct {
+	cell.In
+
+	Logger    logrus.FieldLogger
+	Lifecycle hive.Lifecycle
+
+	ModulesMgr *modules.Manager
+
+	Cfg       Config
+	SharedCfg SharedConfig
+}
+
+func newIptablesManager(p params) *Manager {
+	iptMgr := &Manager{
+		logger:        p.Logger,
+		modulesMgr:    p.ModulesMgr,
+		sharedCfg:     p.SharedCfg,
+		haveIp6tables: true,
+	}
+
+	p.Lifecycle.Append(iptMgr)
+
+	return iptMgr
+}
+
+// Start initializes the iptables manager and checks for iptables kernel modules availability.
+func (m *Manager) Start(ctx hive.HookContext) error {
+	if err := enableIPForwarding(); err != nil {
+		log.Fatalf("enabling IP forwarding via sysctl failed: %s", err)
+	}
+
+	if err := m.modulesMgr.FindOrLoadModules(
+		"ip_tables", "iptable_nat", "iptable_mangle", "iptable_raw", "iptable_filter",
+	); err != nil {
+		m.logger.WithError(err).Warning(
 			"iptables modules could not be initialized. It probably means that iptables is not available on this system")
 	}
-	if err := modulesManager.FindOrLoadModules(
-		"ip6_tables", "ip6table_mangle", "ip6table_raw", "ip6table_filter"); err != nil {
-		if option.Config.EnableIPv6 {
-			log.WithError(err).Fatal(
+
+	if err := m.modulesMgr.FindOrLoadModules(
+		"ip6_tables", "ip6table_mangle", "ip6table_raw", "ip6table_filter",
+	); err != nil {
+		if m.sharedCfg.EnableIPv6 {
+			return errors.New(
```

The error is now dropped, I would wrap it instead with fmt.Errorf and `%w`.